src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -601,5 +601,10 @@ public static string GetFunctionsExtensionVersion(this IEnvironment environment)
         {
             return environment.GetEnvironmentVariableOrDefault(FunctionsExtensionVersion, string.Empty);
         }
+
+        public static bool IsTargetBasedScalingEnabled(this IEnvironment environment)
+        {
+            return string.Equals(environment.GetEnvironmentVariable(TargetBaseScalingEnabled), "1");
```

Comment:
This should check for both App Setting and Hosting Configuration?

Reply:
HostingConfig is checked here:
https://github.com/Azure/azure-functions-host/pull/8780/files#diff-70d41caa568fa2a59c53884cf200b3f224fae1d27962367f9b6b151cd7135685R1028